dags/process_elife_zip_dag.py

```diff
@@ -113,47 +89,90 @@ def extract_archived_files_to_bucket(**context):
                 )
 
         # check if expected article in zip file
-        matches = [fn for fn in ZipFile(temp_file).namelist() if article_name in fn]
-        if not matches:
+        if not [fn for fn in ZipFile(temp_zip_file).namelist() if article_name in fn]:
             error_message = '%s not in %s: %s' % (article_name, zip_file_name,
-                                                  ZipFile(temp_file).namelist())
+                                                  ZipFile(temp_zip_file).namelist())
             raise FileNotFoundError(error_message)
 
     # pass article cloud bucket key to next task
     return '%s/%s' % (folder_name, article_name)
 
 
-def wrap_article_in_libero_xml_and_send_to_service(**context):
-    # get xml path passed from previous task
-    previous_task = get_previous_task_name(**context)
-    xml_path = context['task_instance'].xcom_pull(task_ids=previous_task)
-    logger.info('XML PATH PASSED FROM PREVIOUS TASK= %s', xml_path)
-    message = 'path to xml document was not passed from task %s' % previous_task
-    assert xml_path is not None, message
+def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context):
+    zip_file_name = get_file_name_passed_to_dag_run_conf_file(**context)
+    for key in list_bucket_keys_iter(Bucket=DESTINATION_BUCKET, Prefix=zip_file_name.rstrip('.zip')):
+        if key.endswith('.tif'):
+            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_tiff_file:
+                s3 = get_aws_connection('s3')
+                s3.download_fileobj(
+                    Bucket=DESTINATION_BUCKET,
+                    Key=key,
+                    Fileobj=temp_tiff_file
+                )
+                image = Image.open(temp_tiff_file)
+                output = image.convert('RGB')
+                temp_jpeg = BytesIO()
+                output.save(temp_jpeg, 'JPEG')
+                key = re.sub(r'\.\w+$', '.jpg', key)
```

Comment:
I like the generic match in the regex, allowing other extensions to be introduced easily